src/components/Form/Widgets/FileWidget.tsx

```diff
@@ -0,0 +1,216 @@
+				};
+				reader.onload = () => {
+					if (file != null && reader.result != null) {
+						setIsLoading(false);
+						onChange({
+							dataUrl: reader.result as string,
+							uploadedFile: file,
+						});
+					}
+				};
+
+				reader.readAsDataURL(file);
+			});
+		}) as NonNullable<DropzoneOptions['onDrop']>,
+		[],
+	);
+
+	const { getRootProps, getInputProps, isDragActive } = useDropzone({
+		onDrop,
+		maxSize,
+		accept,
+		multiple: false,
+		noDrag: false,
+		disabled: isLoading,
+	});
+
+	return (
+		<>
+			{props.schema.title && <InputLabel>{props.schema.title}</InputLabel>}
+			<Box
+				sx={{
+					border: (theme) =>
+						`dashed 2px ${
+							errors.length !== 0
+								? '#F19B9E'
+								: isDragActive
+								? theme.palette.primary.main
+								: 'rgba(0, 0, 0, 0.23)'
+						}`,
+					borderRadius: '2px',
+					display: 'flex',
+					flexDirection: 'column',
+					alignItems: 'center',
+					justifyContent: 'center',
+				}}
+				style={{ minHeight: '150px' }}
+				flexDirection="column"
+				alignItems="center"
+				justifyContent="center"
+				{...getRootProps()}
+				dir={undefined}
+			>
+				<input {...getInputProps()} />
+
+				{isLoading && (
+					<Box
+						sx={{
+							position: 'absolute',
+							top: 0,
+							left: 0,
+							width: '100%',
+							height: '100%',
+							display: 'flex',
+							alignItems: 'center',
+							justifyContent: 'center',
+							background: 'rgba(255, 255, 255, 0.7)',
+						}}
+					>
+						<CircularProgress variant="determinate" value={loadingPercentage} />
+					</Box>
+				)}
+
+				<Box display="flex" alignItems="center" justifyContent="center">
+					<Box>
+						{value ? (
+							isImage(value) ? (
+								<Avatar
+									sx={{
+										borderRadius: 0,
+										height: '100px',
+										width: '100px',
+										backgroundSize: 'contain',
+									}}
+									src={dataUrl(value)}
+								/>
+							) : (
+								<FilePresentIcon
+									sx={{
+										fontSize: 64,
+										color: (theme) => theme.palette.grey[500],
+									}}
+								/>
+							)
+						) : (
+							<UploadFileIcon
+								sx={{ fontSize: 64, color: (theme) => theme.palette.grey[500] }}
+							/>
+						)}
+					</Box>
+					<Box>
+						<Typography>
+							<Typography variant="inherit">Drag and drop or</Typography>{' '}
+							<Typography variant="inherit">
+								<Link>browse</Link> files
+							</Typography>
+						</Typography>
+					</Box>
+				</Box>
+			</Box>
+
+			{errors.length !== 0 && (
+				<List>
+					{errors.map((err) => (
+						<ListItem>
+							<ListItemText
+								sx={{ m: -2, color: (theme) => theme.palette.error.main }}
+							>
+								{err}
+							</ListItemText>
+						</ListItem>
+					))}
+				</List>
+			)}
+			{errors.length === 0 && value && typeof value !== 'string' && (
+				<List>
+					<ListItem>
+						<ListItemText sx={{ m: -2 }}>
+							<Typography component="span" variant="body1" fontWeight="bold">
```

Comment:
Not sure if the `component='span'` and `variant='body1'` are both necessary, could you confirm please?

Reply:
Only the span one is needed, thanks